test/users/index.test.js

```diff
@@ -0,0 +1,107 @@
+const supertest = require('supertest');
```

Comment:
Excelent It's a Good Job. I think `that` then in request it's not necessary because you could use `beforeAll` and `done with async`. That allows to finish the test and execute the asserts

I propose you that the testing structure be:


```
const request = require('supertest');

 describe('TESTING POST /users', () => {
    describe('Shoudl be create a new user successful', () => {
          let response = {};
          
          beoforeAll(async done => {
                 responseo = await request(app).post('/users').send(mockUserSuccess)
                 
                 done();
          });
          
          it('The status code must be 201', () => {
                expect(response.statusCode).toBe(201);
          });
          
          it('the response text must be User name has been created successfully', () => {
               expect(response.text).toBe(`User [${mockUserSuccess.name}] has been created succesfully`);
          });
    });
    
    describe('Should be fail when name is missing', () => {
         let response = {};
         beforeAll(async done => {
              response = request(app).post('/users').send(mockUserMissingName);
              
              done();
         });
         
         it('The status code must be 400', () => {
             expect(response.statusCode).toBe(400);
         });
         
         it('The internal error code must be bad_request', () => {
              expect(response.body.internal_code).toBe('bad_request')
         });
    })
};
```

bad_request is a example the middleware validator should be throw like httpClientError, defaultError, notFoundError